modify-publication/src/test/java/no/unit/nva/publication/modify/ModifyPublicationHandlerTest.java

```diff
@@ -123,61 +155,91 @@ public void handlerReturnsBadGatewayResponseOnCommunicationProblems()
             .thenThrow(ErrorResponseException.class);
 
         modifyPublicationHandler.handleRequest(
-            inputStream(publication.getIdentifier().toString()), output, context);
+            generateInputStreamWithValidBodyAndHeadersAndPathParameters(publication.getIdentifier()), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<Problem> gatewayResponse = toGatewayResponseProblem();
         assertEquals(SC_BAD_GATEWAY, gatewayResponse.getStatusCode());
+        assertThat(getProblemDetail(gatewayResponse), containsString(DEFAULT_ERROR_MESSAGE));
     }
 
+
     @Test
     @DisplayName("handler Returns InternalServerError Response On Unexpected Exception")
     public void handlerReturnsInternalServerErrorResponseOnUnexpectedException()
         throws IOException, ApiGatewayException {
         Publication publication = createPublication();
-        when(publicationService.getPublication(any(UUID.class)))
-            .thenReturn(publication);
         when(publicationService.updatePublication(any(UUID.class), any(Publication.class)))
-            .thenThrow(NullPointerException.class);
+            .thenThrow(RuntimeException.class);
 
         modifyPublicationHandler.handleRequest(
-            inputStream(publication.getIdentifier().toString()), output, context);
+            generateInputStreamWithValidBodyAndHeadersAndPathParameters(publication.getIdentifier()), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<Problem> gatewayResponse = toGatewayResponseProblem();
```

Comment:
Here you get a problem because your mock throws a NullPointerException when getPublication() is called before updatePublication